operator/internal/controller/pulumi/git.go

```diff
@@ -18,318 +18,276 @@ import (
-	// 				}
-	// 				gitAuth.Password = password
-	// 			}
-
-	// 			return gitAuth, nil
-	// 		}
-
-	// 		if sess.stack.GitAuth.PersonalAccessToken != nil {
-	// 			accessToken, err := sess.resolveResourceRef(ctx, sess.stack.GitAuth.PersonalAccessToken)
-	// 			if err != nil {
-	// 				return nil, fmt.Errorf("resolving gitAuth personal access token: %w", err)
-	// 			}
-	// 			gitAuth.PersonalAccessToken = accessToken
-	// 			return gitAuth, nil
-	// 		}
-
-	// 		if sess.stack.GitAuth.BasicAuth == nil {
-	// 			return nil, errors.New("gitAuth config must specify exactly one of " +
-	// 				"'personalAccessToken', 'sshPrivateKey' or 'basicAuth'")
-	// 		}
-
-	// 		userName, err := sess.resolveResourceRef(ctx, &sess.stack.GitAuth.BasicAuth.UserName)
-	// 		if err != nil {
-	// 			return nil, fmt.Errorf("resolving gitAuth username: %w", err)
-	// 		}
-
-	// 		password, err := sess.resolveResourceRef(ctx, &sess.stack.GitAuth.BasicAuth.Password)
-	// 		if err != nil {
-	// 			return nil, fmt.Errorf("resolving gitAuth password: %w", err)
-	// 		}
-
-	// 		gitAuth.Username = userName
-	// 		gitAuth.Password = password
-	// 	} else if sess.stack.GitAuthSecret != "" {
-	// 		namespacedName := types.NamespacedName{Name: sess.stack.GitAuthSecret, Namespace: sess.namespace}
-
-	// 		// Fetch the named secret.
-	// 		secret := &corev1.Secret{}
-	// 		if err := sess.kubeClient.Get(ctx, namespacedName, secret); err != nil {
-	// 			sess.logger.Error(err, "Could not find secret for access to the git repository",
-	// 				"Namespace", sess.namespace, "Stack.GitAuthSecret", sess.stack.GitAuthSecret)
-	// 			return nil, err
-	// 		}
-
-	// 		// First check if an SSH private key has been specified.
-	// 		if sshPrivateKey, exists := secret.Data["sshPrivateKey"]; exists {
-	// 			gitAuth = &auto.GitAuth{
-	// 				SSHPrivateKey: string(sshPrivateKey),
-	// 			}
-
-	// 			if password, exists := secret.Data["password"]; exists {
-	// 				gitAuth.Password = string(password)
-	// 			}
-	// 			// Then check if a personal access token has been specified.
-	// 		} else if accessToken, exists := secret.Data["accessToken"]; exists {
-	// 			gitAuth = &auto.GitAuth{
-	// 				PersonalAccessToken: string(accessToken),
-	// 			}
-	// 			// Then check if basic authentication has been specified.
-	// 		} else if username, exists := secret.Data["username"]; exists {
-	// 			if password, exists := secret.Data["password"]; exists {
-	// 				gitAuth = &auto.GitAuth{
-	// 					Username: string(username),
-	// 					Password: string(password),
-	// 				}
-	// 			} else {
-	// 				return nil, errors.New("creating gitAuth: missing 'password' secret entry")
-	// 			}
-	// 		}
-	// 	}
-
-	// 	if u.Scheme == "ssh" && gitAuth.SSHPrivateKey == "" {
-	// 		return gitAuth, fmt.Errorf("a private key must be provided for SSH")
-	// 	}
-
-	// 	return gitAuth, nil
+// Source represents a source of commits.
+type Source interface {
+	CurrentCommit(context.Context) (string, error)
```

Comment:
Is `Source` an interface that the Flux stuff should also use?

Reply:
Only if it would be helpful for the purpose of mocking GitHub/Flux in unit tests -- otherwise we don't need it.